src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/ApartmentFactory.java

```diff
@@ -198,46 +262,65 @@ private static String getRandomAddress(int retry) throws IOException {
 	 */
 	private static String getRandomAddress() {
 		try {
-			return getRandomAddress(0);
-		} catch (IOException e) {
-			LOGGER.error("Problem while getting random address", e);
+			return ApartmentFactory.getOnlineRandomAddress();
+		} catch (ClientErrorException | AddressApiException e) {
+			// We return an apipa address
+			LOGGER.error("Problem while getting random address {}", e.toString());
 			StringBuilder sb = new StringBuilder();
-			sb.append(rand.nextInt(3000)).append(" rue de l'appel échoué ").append(rand.nextInt(19) + 75001)
-			.append(" Paris ");
+			sb.append(ApartmentFactory.rand.nextInt(3000)).append(" rue de l'appel échoué ")
+					.append(ApartmentFactory.rand.nextInt(19) + 75001).append(" Paris ");
 			return sb.toString();
 		}
 	}
 
 	/**
-	 * Call an API which generates an existing random address. This function aims at
-	 * getting the random address generated.
-	 * 
-	 * @return the random address
-	 * @throws IOException if the Address API cannot answer
-	 */
-	private static String getRandomAddressOnlyReal() throws IOException {
-		return getRandomAddress(0);
-	}
-
-	/**
-	 * Generate a list of apartments from a json file.
+	 * Generate a list of apartments from a JSON file.
 	 *
-	 * @param jsonFileAddress the address where the json with all the apartments are
-	 * @return the list of apartments found in the json file.
-	 * @throws IOException if we cannot have access to the json file.
+	 * @param jsonFilePath the address where the JSON with all the apartments are
+	 * @return the list of apartments found in the JSON file.
+	 * @throws IOException if we cannot have access to the JSON file.
 	 */
-	public static List<Apartment> generateApartmentFromJsonPath(Path jsonFilePath) throws IOException {
+	public static List<Apartment> generateApartmentsFromJsonPath(Path jsonFilePath) throws IOException {
 		return JsonConvert.jsonToApartments(jsonFilePath);
 	}
 
 	/**
-	 * Generate apartment from a json file by default.
+	 * Generate apartment from a JSON file by default.
 	 *
-	 * @return the list of apartments found in the json file by default.
-	 * @throws IOException if we cannot have access to the json file.
+	 * @return the list of apartments found in the JSON file by default.
+	 * @throws IOException if we cannot have access to the JSON file.
 	 */
 	public static List<Apartment> getDefaultApartments() throws IOException {
```

Comment:
Ceci ne devrait pas renvoyer `IOException`, vous pouvez considérer qu’une lecture depuis le class path n’échoue pas (il n’y a pas plus de raison que d’imaginer que le code de vos classes lui-même ne soit pas accessible, puisqu’il est lu de la même façon, ce qui empêcherait quoi que ce soit de fonctionner).

Reply:
Le `IOException` provient de `Files.readString(...)`
Devons nous `catch` l'exception dans la classe `JsonConvert` ?

Reply:
Convertir l’exception en IllStEx en l’enrobant, là où vous semble approprié (valable pour autant qu’il s’agisse d’une lecture depuis le class path, sinon UncheckedIOE est appropriée).

Reply:
Fix in 6a338d33ebf81607fa41f3766b6108320228291f